common/scala/src/main/scala/whisk/core/connector/Message.scala

```diff
@@ -56,7 +56,8 @@ case class ActivationMessage(
     activationNamespace: EntityPath,
     rootControllerIndex: InstanceId,
     content: Option[JsObject],
-    cause: Option[ActivationId] = None)
+    cause: Option[ActivationId] = None,
+    tracingMetadata: Option[Map[String, String]] = None)
```

Comment:
In case of jaeger it's one key-value pair, where key is "jaeger-id" (or similar) and value some short string, zipkin is similar. So it's couple of bytes. What can also help is setting the sampling to some lower value, like 5% so that only random 5% of all actions will be traced (which reminded me that I should probably set the [default value](https://github.com/apache/incubator-openwhisk/pull/2413/files#diff-1048d039011bd0a7de7d2763733749e8R41) to something more sane, like 10% and perhaps make it configurable in the wisk.properties (?))

>for data, I expect that message to flow externally through a redis for example.

not sure I follow you here, are you planning to add also redis in the future?